cmd/cost/get.go

```diff
@@ -23,48 +24,46 @@ func newCmdGet(streams genericclioptions.IOStreams) *cobra.Command {
 		Use:   "get",
 		Short: "Get total cost of a given OU",
 		Run: func(cmd *cobra.Command, args []string) {
-
-			awsClient, err := opsCost.initAWSClients()
-			cmdutil.CheckErr(err)
-
-			//Get information regarding Organizational Unit
-			OU := getOU(awsClient, ops.ou)
-
-			var cost float64
-			var unit string
-
-			if ops.recursive { //Get cost of given OU by aggregating costs of all (including immediate) accounts under OU
-				if err := getOUCostRecursive(&cost, &unit, OU, awsClient, &ops.time); err != nil {
-					log.Fatalln("Error getting cost of OU recursively:", err)
-				}
-			} else { //Get cost of given OU by aggregating costs of only immediate accounts under given OU
-				if err := getOUCost(&cost, &unit, OU, awsClient, &ops.time); err != nil {
-					log.Fatalln("Error getting cost of OU:", err)
-				}
-			}
-			printCostGet(cost, unit, ops, OU)
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
 		},
 	}
 	getCmd.Flags().StringVar(&ops.ou, "ou", "", "get OU ID")
 	getCmd.Flags().BoolVarP(&ops.recursive, "recursive", "r", false, "recurse through OUs")
 	getCmd.Flags().StringVarP(&ops.time, "time", "t", "", "set time. One of 'LM', 'MTD', 'TYD', '3M', '6M', '1Y'")
+	getCmd.Flags().StringVar(&ops.start, "start", "", "set start date range")
+	getCmd.Flags().StringVar(&ops.end, "end", "", "set end date range")
 	getCmd.Flags().BoolVar(&ops.csv, "csv", false, "output result as csv")
 
-	if err := getCmd.MarkFlagRequired("ou"); err != nil {
-		log.Fatalln("OU flag:", err)
+	return getCmd
+}
+
+func (o *getOptions) complete(cmd *cobra.Command, _ []string) error {
```

Comment:
This function name is not very clear to me. Is there a reason its named complete and not something like "checkArgs"? Could you add some comments to this as well to explain what this function does and failure case?

Reply:
Refactoring this to make it more clear